### ISSUE TITLE
TypeScript errors when running in `npm start` or `npm run build`

### ISSUE THREAD

**Issue by roebuk** [OfficeDev/generator-office#277]
I created a new project using this generator and  the following settings:

```
Would you like to create a new subfolder for your project? Yes
What do you want to name your add-in? OutlookAddIn
Which Office client application would you like to support? Outlook
Would you like to create a new add-in? Yes, I need to create a new web app and manifest file for my add-in.
Would you like to use TypeScript? Yes
Choose a framework: React
```

When I run `npm start` I recieve the following 20 errors:

```
[at-loader] Checking finished with 20 errors
[at-loader] ./node_modules/@types/react/index.d.ts:169:11
    TS2559: Type 'Component<P, S>' has no properties in common with type 'ComponentLifecycle<P, S>'.

[at-loader] ./node_modules/@uifabric/utilities/lib/Customizer.d.ts:22:55
    TS2559: Type 'ICustomizerProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/Calendar/CalendarDay.d.ts:30:56
    TS2559: Type 'ICalendarDayProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/ColorPicker/ColorPicker.d.ts:8:56
    TS2559: Type 'IColorPickerProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/DetailsList/DetailsHeader.d.ts:42:58
    TS2559: Type 'IDetailsHeaderProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/DetailsList/DetailsRow.d.ts:44:55
    TS2559: Type 'IDetailsRowProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/Dialog/DialogContent.d.ts:6:58
    TS2559: Type 'IDialogContentProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/DocumentCard/DocumentCardActions.d.ts:3:64
    TS2559: Type 'IDocumentCardActionsProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/DocumentCard/DocumentCardActivity.d.ts:3:65
    TS2559: Type 'IDocumentCardActivityProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/DocumentCard/DocumentCardLocation.d.ts:3:65
    TS2559: Type 'IDocumentCardLocationProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/DocumentCard/DocumentCardPreview.d.ts:3:64
    TS2559: Type 'IDocumentCardPreviewProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/DocumentCard/DocumentCardTitle.d.ts:7:62
    TS2559: Type 'IDocumentCardTitleProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/Fabric/Fabric.d.ts:6:51
    TS2559: Type 'HTMLProps<HTMLDivElement>' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/Layer/LayerHost.d.ts:3:54
    TS2559: Type 'HTMLProps<HTMLElement>' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/Pivot/PivotItem.d.ts:3:54
    TS2559: Type 'IPivotItemProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/pickers/AutoFill/BaseAutoFill.d.ts:6:57
    TS2559: Type 'IBaseAutoFillProps' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/pickers/BasePicker.d.ts:18:89
    TS2559: Type 'P' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/pickers/Suggestions/Suggestions.d.ts:4:63
    TS2559: Type 'ISuggestionItemProps<T>' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/components/pickers/Suggestions/Suggestions.d.ts:7:59
    TS2559: Type 'ISuggestionsProps<T>' has no properties in common with type 'IBaseProps'.

[at-loader] ./node_modules/office-ui-fabric-react/lib/utilities/selection/SelectionZone.d.ts:14:58
    TS2559: Type 'ISelectionZoneProps' has no properties in common with type 'IBaseProps'.
```

This also prevents me from being able to sucessfully run the `npm run build` command. The build step bails out.

```
Error in bail mode: [at-loader] ./node_modules/@types/react/index.d.ts:169:11
    TS2559: Type 'Component<P, S>' has no properties in common with type 'ComponentLifecycle<P, S>'.
```

**Comment by roebuk** [comment]
Update:

I've bumped to the latest versions of `typescript`, `office-ui-fabric-react`, and `@types/office-js` the issue still persists. Applogies for not diving into this any further this is the first project I've dived into with TypeScript.

**Comment by dimamarksman** [comment]
The same for me

**Comment by dimamarksman** [comment]
I updated dependencies and now it works
```
    "dependencies": {
        "core-js": "^2.5.3",
        "office-addin-validator": "^1.0.5",
        "office-ui-fabric-react": "^5.44.0",
        "react": "^15.4.2",
        "react-dom": "^15.4.2",
        "typescript": "^2.2.1"
    },
    "devDependencies": {
        "@types/office-js": "0.0.51",
        "@types/react": "16.0.35",
        "@types/react-dom": "16.0.3",
        "@types/webpack": "2.2.12",
        "@types/webpack-dev-server": "1.12.6",
        "autoprefixer": "6.7.7",
        "awesome-typescript-loader": "3.1.2",
        "browser-sync": "2.18.8",
        "browser-sync-webpack-plugin": "1.1.4",
        "copy-webpack-plugin": "4.0.1",
        "css-loader": "0.27.3",
        "extract-text-webpack-plugin": "2.1.0",
        "file-loader": "0.10.1",
        "html-webpack-plugin": "2.28.0",
        "node-sass": "^4.7.2",
        "perfectionist": "2.4.0",
        "postcss-loader": "1.3.3",
        "precss": "1.4.0",
        "react-hot-loader": "1.3.1",
        "rimraf": "2.6.1",
        "sass-loader": "6.0.3",
        "style-loader": "0.16.0",
        "tslint": "4.5.1",
        "webpack": "2.3.1",
        "webpack-dev-server": "2.4.2",
        "webpack-merge": "4.1.0"
    }
```

**Comment by JoyPeterson** [comment]
After I update to those version I get the following error:

```
Error in bail mode: [at-loader] ./src/components/app.tsx:56:89
    TS2339: Property 'icon' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<Button> & Readonly<{ children?: ReactNode; }> & Re...'.
```

Any ideas?

**Comment by dimamarksman** [comment]
The example code is outdated here.
Please see the current [spec](https://developer.microsoft.com/en-us/fabric#/components/button) how it should be used properly.

**Comment by LouMM** [comment]
We are going to drastically simplify the Templates for yo Office. This should help with the versioning issues and the outdated libraries. Look out for a PR near you, soon. Thanks to @JoyPeterson , @dimamarksman  and @roebuk  for the great reports!

**Comment by LouMM** [comment]
Thank you to everyone for your input for giving offline assistance in providing data surrounding these issues. I think we got it and fixed it in #280 . New NPM packages should be promoted soon!

**Comment by LouMM** [comment]
Quick update, NPM package has been updated to 1.1.10 and has these items fixed.
